src/incubator/TextField/useFieldState.ts

```diff
@@ -42,7 +42,7 @@ export default function useFieldState({
     if (props.value !== value) {
       setValue(props.value);
 
-      if (validateOnChange) {
+      if (validateOnChange && (!Constants.isWeb || value !== props.defaultValue)) {
```

Comment:
From what I've seen, it also reproduces on mobile
Please verify it and if it's true, add this condition constantly :)